src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
Replace .forEach() with .reduce() for calculating the total age difference. This avoids the need for explicit looping.

```suggestion
  const total = children.reduce((sum, child) => {
    return sum + (child.born - peopleByName[child.mother].born);
  }, 0);

  return total / children.length;
```